Library/Homebrew/test/formulary_spec.rb

```diff
@@ -0,0 +1,215 @@
+require "formula"
+require "formula_installer"
+require "utils/bottles"
+
+describe Formulary do
+  let(:name) { "testball_bottle" }
+  let(:path) { CoreTap.new.formula_dir/"#{name}.rb" }
+  let(:content) {
+    <<-EOS.undent
+      class #{described_class.class_s(name)} < Formula
+        url "file://#{TEST_FIXTURE_DIR}/tarballs/testball-0.1.tbz"
+        sha256 TESTBALL_SHA256
+
+        bottle do
+          cellar :any_skip_relocation
+          root_url "file://#{bottle_dir}"
+          sha256 "9abc8ce779067e26556002c4ca6b9427b9874d25f0cafa7028e05b5c5c410cb4" => :#{Utils::Bottles.tag}
+        end
+
+        def install
+          prefix.install "bin"
+          prefix.install "libexec"
+        end
+      end
+    EOS
+  }
+  let(:bottle_dir) { Pathname.new("#{TEST_FIXTURE_DIR}/bottles") }
+  let(:bottle) { bottle_dir/"testball_bottle-0.1.#{Utils::Bottles.tag}.bottle.tar.gz" }
+
+  specify "::class_s" do
```

Comment:
I think this should be split into multiple examples with descriptive names, e.g. `it "replaces '+' with 'x'"`.